devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,554 @@
+
+		// Viconics VT8650 Heat Pump and Indoor Air Quality Controller
+		// Raw Description 0A 0104 0301 00 09 0201 0405 0402 0406 0204 0000 0004 0003 0005 0A 0201 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Viconics", model: "254-162", deviceJoinName: "Viconics Room Controller", mnmn: "SmartThings", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller"
+		//fingerprint profileId: "0104", inClusters: "0000,0003,0201,0204,0405", outClusters: "0402,0405", manufacturer: "Viconics", model: "254-162", deviceJoinName: "VT8650xx"
+
+		// Schneider Electric SE8350 Low Voltage Fan Coil Unit (FCU) and Zone Control
+		// Raw Description 0A 0104 0301 00 0A 0201 0202 0405 0402 0406 0204 0000 0004 0003 0005 0B 0201 0202 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Schneider Electric", model: "254-145", deviceJoinName: "Schneider Electric Room Controller", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller_Fan"
+
+		// Schneider Electric SE8650 Roof Top Unit Controller
+		// Raw Description 0A 0104 0301 00 09 0201 0405 0402 0406 0204 0000 0004 0003 0005 0A 0201 0405 0402 0406 0204 0000 0004 0003 0005 0500
+		fingerprint manufacturer: "Schneider Electric", model: "254-163", deviceJoinName: "Schneider Electric Room Controller", vid: "SmartThings-smartthings-Viconics_Schneider_Room_Controller"
+	}
+}
+
+private getTHERMOSTAT_CLUSTER() { 0x0201 }
+private getTHERMOSTAT_UI_CONFIGURATION_CLUSTER() { 0x0204 }
+private getTEMPERATURE_DISPLAY_MODE() {0x0000}
+private getLOCAL_TEMPERATURE() {0x0000}
+private getCOOLING_SETPOINT() { 0x0011 }
+private getHEATING_SETPOINT() { 0x0012 }
+private getCOOLING_SETPOINT_UNOCCUPIED() { 0x0013 }
+private getHEATING_SETPOINT_UNOCCUPIED() { 0x0014 }
+private getOCCUPANCY() { 0x002 }
+private getCUSTOM_OCCUPANCY() {0x0650}
+private getCUSTOM_EFFECTIVE_OCCUPANCY() { 0x0c50 }
+private getCUSTOM_HUMIDITY() { 0x07a6 }
+private getCUSTOM_THERMOSTAT_MODE() { 0x0687 }
+private getCUSTOM_FAN_SPEED() { 0x0688 }
+private getCUSTOM_FAN_MODE() { 0x0698 }
+private getCUSTOM_THERMOSTAT_OPERATING_STATE() { 0x06BF }
+private getUNOCCUPIED_SETPOINT_CHILD_DEVICE_ID() {1}
+private getTHERMOSTAT_MODE_OFF() { 0x00 }
+private getTHERMOSTAT_MODE_AUTO() { 0x01 }
+private getTHERMOSTAT_MODE_COOL() { 0x02 }
+private getTHERMOSTAT_MODE_HEAT() { 0x03 }
+private getCUSTOM_FAN_MODE_ON() { 0x00 }
+private getCUSTOM_FAN_MODE_AUTO() { 0x01 }
+private getCUSTOM_FAN_MODE_CIRCULATE() { 0x02 }
+private getOPERATING_STATE_IDLE() { 0x00 }
+private getOPERATING_STATE_COOLING() { 0x01 }
+private getOPERATING_STATE_HEATING() { 0x02 }
+private getOCCUPANCY_OCCUPIED() { 0x00 }
+private getOCCUPANCY_UNOCCUPIED() { 0x01 }
+private getFAN_SPEED_LOW() { 0x00 }
+private getFAN_SPEED_MEDIUM() { 0x01 }
+private getFAN_SPEED_HIGH() { 0x02 }
+private getFAN_SPEED_AUTO() { 0x03 }
```

Comment:
I think it might be worthwhile to look into using an enum for some of these.

Reply:
@greens 
wouldn't that look bad in the code if some of those were enums and some not ? 
Old way (declaration and usage) looks more concise.

enums for occupancy might look like this:

```
enum Occupancy{
	OCCUPIED(0), UNOCCUPIED(1)
	Occupancy(int value) {
		this.value = value
	}
	private final int value
	int getValue() {
		value
	}
}
```

Reply:
yeah fair enough